pennylane/_device.py

```diff
@@ -175,6 +175,26 @@ def observables(self):
         """
         raise NotImplementedError
 
+    @property
+    def shots(self):
+        """Number of circuit evaluations/random samples used to estimate
+            expectation values of observables"""
+
+        return self._shots
+
+    @shots.setter
+    def shots(self, shots):
+        """Changes the number of shots.
+
+        Args:
+            shots (int): number of circuit evaluations/random samples used to estimate
+            expectation values of observables
+        """
+        if shots < 1:
+            raise DeviceError("The specified number of shots needs to be at least 1. Got {}.".format(shots))
+
+        self._shots = int(shots)
```

Comment:
:+1: I think this is the best approach, because a float will be silently promoted to a integer, whereas anything else should fail with an exception here, rather than failing with a more cryptic error message later.

Reply:
:+1: